### DISCUSSION TITLE
[REQUEST] Table or Array component

### DISCUSSION THREAD

**Discussion by KarstenTWolf** [FakeFishGames/Barotrauma#5025] [Ideas and suggestions]
With the new update the CONC component has a max output of 256, which breaks a use case where the CONC was used to append a large amount of lookup data to a search term as a sort of table lookup using a regex.

A table or array component would solve this functionality while not breaking a 256 character output limit. Individual cells could be limited to 256 characters but could be referenced by an index # input.

A .csv or similar import function would be amazing, as importing this dataset is probably the most editor/ingame unfriendly part with the current UI.

One counter argument is that this could be done with a number of sig check and OR components, but having this in a single component could greatly simplify anything that uses lots of text such as terminal UI, chat interactions or labels.

One of my item assemblies would require around 50 MEM or OR components to replicate what was done previously with a single CONC before the 256 output limit.

**Comment by Xeavas** [comment]
@KarstenTWolf you can increase the max string output to a value of your choice (in sub-editor).

**Reply by KarstenTWolf** [reply]
Ok, but I still want an array/table component even if my tantrum was unfounded. =P
